### PR TITLE
Mar3hc/feat/write data by name

### PR DESCRIPTION
**Pull request by trimai3001**

### CODE REVIEW

File: test/example.robot [hunk 1]
```diff
@@ -77,4 +75,11 @@ Test user can use Routine Control By Name service on ECU
...
-    Diagnostic Session Control    1
\ No newline at end of file
+    Diagnostic Session Control    1
+
+Test user can use Write Data By Name service on ECU
+    Log    Write Data By Name service
+
+    Log    RealTimeClock_Write
+    ${PARAM_DICT}=    Create Dictionary    Day=26    Month=September    Year=2024    Hour=10    Second=45    Minute=0
+    Write Data By Name    RealTimeClock_Write    ${PARAM_DICT}
```
Comment (ngoan1608) [inline test/example.robot:85]: Hi Tri,

Please also add the test case for keyword `Write Data By Identifier` as we discussed.

Thank you,
Ngoan

File: test/example.robot [hunk 2]
```diff
@@ -13,9 +13,7 @@ ${ACTIVATION_TYPE}=    0
 
 ${FILE}=    C:/Users/MAR3HC/Desktop/UDS/robotframework-uds/test/pdx/CTS_STLA_V1_15_2.pdx
 ${VARIANT}=    CTS_STLA_Brain
-
-${NAME}=    UDS Connector
-
+${PARAM_STR}=    Day=26, Month=September, Year=2024, Hour=10, Second=45, Minute=0
```
Comment (ngoan1608) [inline test/example.robot]: This `${PARAM_STR}` is not used, consider to remove it

File: RobotFramework_UDS/UDSKeywords.py [hunk 3]
```diff
@@ -1155,8 +1154,19 @@ def write_data_by_identifier(self, did: int, value: Any, device_name="default"):
 
   The response from the WriteDataByIdentifier service request.
         """
+        logger.info(f"Service DID: {did}")
         uds_device = self.__device_check(device_name)
+        SID_RQ = 46 # The request id of read data by identifier
```
Comment (ngoan1608) [inline RobotFramework_UDS/UDSKeywords.py]: `SID = 46` is write data, not read

**Review by ngoan1608 — CHANGES_REQUESTED** [review]
Hi @trimai3001 ,

Please refer my review finding and provide the proper updates.

Thank you,
Ngoan

**Review by test-fullautomation — CHANGES_REQUESTED** [review]
Hi Tri,

I wait till you commit the changes requested by Ngoan.

Thank you,
 Thoma

**Comment by trimai3001** [comment]
Hi Thomas,

I've resolved all of Ngoan's review findings. Please let me know if I missed anything.

Thank you,
Tri

**Review by ngoan1608 — APPROVED** [review]

**Review by test-fullautomation — APPROVED** [review]
Hi Tri,
very good!
Thank you,
 Thomas

**Pull request merged**